robotoff/prediction/ocr/packager_code.py

```diff
@@ -34,75 +34,136 @@ def process_fsc_match(match) -> str:
     return "FSC-{}".format(fsc_code).upper()
 
 
-PACKAGER_CODE: Dict[str, OCRRegex] = {
-    "fr_emb": OCRRegex(
-        re.compile(r"emb ?(\d ?\d ?\d ?\d ?\d) ?([a-z])?(?![a-z0-9])"),
-        field=OCRField.text_annotations,
-        lowercase=True,
-        processing_func=process_fr_emb_match,
-    ),
-    "fsc": OCRRegex(
-        re.compile(r"fsc.? ?(c\d{6})"),
-        field=OCRField.text_annotations,
-        lowercase=True,
-        processing_func=process_fsc_match,
-    ),
-    "eu_fr": OCRRegex(
-        re.compile(
-            r"fr (\d{2,3}|2[ab])[\-\s.](\d{3})[\-\s.](\d{3}) (ce|ec)(?![a-z0-9])"
+def process_USDA_match_to_flashtext(match) -> Optional[Any]:
+    """this functions returns the USDA code matched by REGEX the same way it exists
+    in the USDA database (1st column of USDA_code_flashtext.txt)"""
+
+    unchecked_code = match.group().upper()
+    unchecked_code = re.sub(r"\s*\.*", "", unchecked_code)
+
+    processor = USDA_CODE_KEYWORD_PROCESSOR_STORE.get()
+    USDA_code = extract_USDA_code(processor, unchecked_code)
+    return USDA_code
+
+
+def generate_USDA_code_keyword_processor() -> KeywordProcessor:
```

Comment:
Please add a docstring to every function :-)
Eg.
```suggestion
def generate_USDA_code_keyword_processor() -> KeywordProcessor:
    """Build the KeyWordProcessor for USDA codes
    
    This will be called only once thanks to CachedStore
    """
```